### PR TITLE
minigame#2 final mechanics, task #23

### PR DESCRIPTION
**Pull request by Starnaphie**
added tutorial popup for minigame2, changed all of the alerts to popups after each question, linked final tutorial popup to minigame end screen
- cleaned up minigame-end link in minigame1, no longer needed as an example with the implementation in minigame2
- popup implementation in budget-game-setup for dynamic title and content to show up in index
- added tests for edge cases of Popup component

next steps
- minigame2 star implementation 
- UI updates

or 
> npm test

necessary dependencies for testing include:
- vitest testing library
- general react testing library
- the code should already include: json package scripts to include "test: "vitest"", vitest config file, setuptest configs

### CODE REVIEW

**Comment by Ydz0616** [comment]
it'd be nice and a good practice to remove package lock as it may contain sensitive data

File: src/minigames/minigame1-scholarship/index.tsx
Comment (Ydz0616) [inline src/minigames/minigame1-scholarship/index.tsx:1]: it'd be nice to include one comment in commit message to explain why Minigame result is removed for game1 and added for game 2( conflicts in game 1, other stuff? ) ;

also, looking forward to talk about result logic in tomorrow's standup :-) maps game status ( green|red|grey) would need that

File: src/minigames/minigame2-budgeting/budget-game-setup.ts
Comment (Ydz0616) [inline src/minigames/minigame2-budgeting/budget-game-setup.ts:1]: this is nice, i could see you've replaced the alert placeholder with actual popup! the edit is clear and the comments are clear; 

It'd be nice to add a few lines of comments to describe the popup ( e.g. does it require user interaction? or it just pops up for a few seconds ; what does the user need to do? )

it'd be nice to add test cases for the popup ( i'm not worried about the math, that'll be fine,  i'm a bit concerned about two things : 1. popup lifecycle ( does it spawns and destroys itself according to plan ) 2. popup performance ( would it jam the frontend, a common problem for react)

File: src/minigames/minigame2-budgeting/index.tsx [hunk 1]
```diff
@@ -12,17 +14,26 @@ export const metadata = {
...
     currentIncome,
-    submitAnswer
+    submitAnswer,
+    title,
+    content,
+    last
   } = useBudgetGameLogic();
 
+  const result: MinigameResult = {  
+    stars: 1,
```
Comment (Ydz0616) [inline src/minigames/minigame2-budgeting/index.tsx:39]: here could add a comment about why star is by default 1

File: src/minigames/minigame2-budgeting/index.tsx [hunk 2]
```diff
@@ -32,9 +43,23 @@ const Minigame2: React.FC = () => {
     );
   };
 
+  const submitHelper = () => {
+    submitAnswer();
+    console.log({title}, {content})
```
Comment (Ydz0616) [inline src/minigames/minigame2-budgeting/index.tsx]: console.log title content is for debugging, right? could comment this out after eyeballing the result is correct ( no need a test, this is very nice) ; or, could add a bit more explanation to this log (e.g. Minigame2: {title},{content}) so it would not confuse other people working on this

File: src/minigames/minigame2-budgeting/index.tsx [hunk 3]
```diff
@@ -32,9 +43,23 @@ const Minigame2: React.FC = () => {
     );
   };
 
+  const submitHelper = () => {
```
Comment (Ydz0616) [inline src/minigames/minigame2-budgeting/index.tsx:53]: nice, very clear, love this helper

File: src/minigames/minigame2-budgeting/index.tsx
Comment (Ydz0616) [inline src/minigames/minigame2-budgeting/index.tsx:1]: awesome job reusing components pop-up! it's designed to be re-used and i'm just glad people could save time using it.

please find my line-specific review/comments below

File: src/tests/index.test.tsx
Comment (Ydz0616) [inline src/tests/index.test.tsx:1]: could consider testing popup lifecycle, front-end jamming, i knew in some cases the chrome's built in modal could jam the browser; a more elaborate testing plan is needed here

File: package-lock.json
Comment (Ydz0616) [inline package-lock.json:1]: please remove this from commit

**Review by Ydz0616 — CHANGES_REQUESTED** [review]
Overall, it is amazing, love how pop-up is reused, and there's clear comment and for most cases the code speaks for itself, so, awesome work!!

i was a bit shocked when i saw +664, then relieved realizing most is package lock -- it is best practice to exclude package lock from commit since it may contain sensitive info about our app

please find other comments for each file.

File: src/minigames/minigame1-scholarship/index.tsx
Comment (Starnaphie) [reply src/minigames/minigame1-scholarship/index.tsx:1]: i'll answer this in more depth in the pr description, but the implementation in minigame1 i removed wasn't official - it was an example implementation i wrote to model other implementations off of. so, now that minigame2 has a official, fleshed out implementation, i went back to remove the one in minigame1 so it can be reimplemented correctly based on what is needed!

File: src/minigames/minigame2-budgeting/budget-game-setup.ts
Comment (Starnaphie) [reply src/minigames/minigame2-budgeting/budget-game-setup.ts:1]: thanks for the feedback! 
i added a description for the Popup in the Popup component file, and test cases for the popup already exist in the generalui tests file (written in the pr for the popup component)
i'll look into the popup performance test case since i'm not sure if the existing ones cover it already...

File: src/minigames/minigame2-budgeting/index.tsx (hunk 1 above)
Comment (Starnaphie) [reply src/minigames/minigame2-budgeting/index.tsx:39]: this implementation is incomplete! i believe Mo is responsible for writing star logic 
(i added a TODO comment for Mo)

File: src/minigames/minigame2-budgeting/index.tsx (hunk 2 above)
Comment (Starnaphie) [reply src/minigames/minigame2-budgeting/index.tsx]: added more detail to console message and commented it out!

File: src/minigames/minigame2-budgeting/index.tsx (hunk 3 above)
Comment (Starnaphie) [reply src/minigames/minigame2-budgeting/index.tsx:53]: thxs ^u^

File: src/tests/index.test.tsx
Comment (Starnaphie) [reply src/tests/index.test.tsx:1]: added!

**Review by Ydz0616 — APPROVED** [review]
looks amazing!

**Review by Maxi-Garcia — APPROVED** [review]
Your changes look good! thanks for modifying the test cases that involved the pop up from my code. Tested and ran all cases on my local machine and it works well.

**Review by malapatiUCSD — APPROVED** [review]
Good job connecting it to that final end screen makes my life so much easier! LGTM approved
